### PR TITLE
plugins.afreeca: rename to soop

### LINKED ISSUE #6246: plugins.afreeca: The afreeca domain has changed to https://www.sooplive.co.kr/
### Checklist

- [x] This is a [plugin issue](https://streamlink.github.io/plugins.html) and not [a different kind of issue](https://github.com/streamlink/streamlink/issues/new/choose)
- [x] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [x] [I have checked the list of open and recently closed plugin issues](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22plugin+issue%22)
- [x] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Collaboration

- [x] [I will provide feedback should a pull request be opened with a fix for the plugin](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#pull-request-feedback)

### Streamlink version

6.11.0

### Description

The afreeca domain has changed to https://www.sooplive.co.kr/. The website has also been rebranded as "sooplive". Although https://www.afreecatv.com/?hash=bora is still accessible, visiting this link will redirect you to https://www.sooplive.co.kr/. The afreeca plugins are still functioning normally, but it is uncertain when they will cease to work.

### Debug log

no log

### PR DESCRIPTION
**Pull request by minibox24**
Resolves #6246

Worked on changing hosts as `afreecatv.com` was rebranded to `sooplive.co.kr`.
Fixed login functionality that was not working due to this.

### CODE REVIEW

**Comment by bastimeyer** [comment]
Thanks for the PR.

Considering that they completely rebranded, it doesn't make sense keeping the "afreeca" plugin name.

So, could you please
1. rename the plugin and its test file from `afreeca` to `sooplive`
2. rename the plugin class, the HLS subclasses and test class to `SoopLive`
3. change the order of host names, both in the plugin module metadata at the top as well as in the plugin-matcher regex, so that afreeca comes last (as it's only a fallback host name now)
4. change the order of test URLs

**Comment by minibox24** [comment]
I've made the changes based on the feedback you gave me. In Korea, it is referred to as 'soop' rather than 'sooplive,' so I changed the plugin name to `soop`.

I was wondering if there's any way to alias the plugin name so that users don't find it inconvenient when they have to change `--afreeca-username` to `--soop-username`?

File: src/streamlink/plugins/soop.py [hunk 1]
```diff
@@ -149,7 +154,9 @@ def _get_hls_stream(self, broadcast, username, quality, rmd, stream_password):
         info = self._get_stream_info(broadcast, quality, rmd)
 
         if "view_url" in info:
-            return AfreecaHLSStream(self.session, info.get("view_url"), params={"aid": key})
+            return SoopHLSStream(
+                self.session, info.get("view_url"), params={"aid": key},
```
Comment (bastimeyer) [inline src/streamlink/plugins/soop.py:158]: Please avoid setting multiple arguments or keywords on the same line.

File: src/streamlink/plugins/soop.py [hunk 2]
```diff
@@ -162,7 +169,9 @@ def _login(self, username, password):
             "isSaveJoin": "false",
             "isLoginRetain": "Y",
         }
-        res = self.session.http.post("https://login.afreecatv.com/app/LoginAction.php", data=data)
+        res = self.session.http.post(
+            "https://login.sooplive.co.kr/app/LoginAction.php", data=data,
```
Comment (bastimeyer) [inline src/streamlink/plugins/soop.py:173]: Same here

File: src/streamlink/plugins/soop.py [hunk 3]
```diff
@@ -222,7 +233,9 @@ def _get_streams(self):
         for item in channel.get("VIEWPRESET"):
             if item["name"] == "auto":
                 continue
-            if hls_stream := self._get_hls_stream(broadcast, username, item["name"], rmd, stream_password):
+            if hls_stream := self._get_hls_stream(
+                broadcast, username, item["name"], rmd, stream_password,
```
Comment (bastimeyer) [inline src/streamlink/plugins/soop.py:237]: And here

**Review by bastimeyer — COMMENTED** [review]
> so I changed the plugin name to `soop`

:+1: 

> I was wondering if there's any way to alias the plugin name so that users don't find it inconvenient when they have to change `--afreeca-username` to `--soop-username`?

Plugin arguments currently don't support aliases (similar to the argparse module). This requires adding duplicates and suppressing them, so they're not listed in the docs, man page and help output. Their values then need to be set in place of the regular arguments if they are missing.

However, the values of suppressed arguments currently get discarded/ignored when set, because there wasn't a use-case for this yet. I've just changed this behavior in #6249 which I just opened. You'll have to rebase your PR onto the new HEAD of master once #6249 got merged in order to test this.

Another limitation of suppressed plugin arguments is that any argument requirements will be ignored, so if only `--afreeca-username` is set, it won't ask for a value of `--afreeca-password`/`--soop-password`.

Here's a diff that would add support for the old arguments:
```diff
diff --git a/src/streamlink/plugins/soop.py b/src/streamlink/plugins/soop.py
index d41eb7e9..073a2d23 100644
--- a/src/streamlink/plugins/soop.py
+++ b/src/streamlink/plugins/soop.py
@@ -8,6 +8,7 @@ $metadata author
 $metadata title
 """
 
+import argparse
 import logging
 import re
 
@@ -60,7 +61,37 @@ class SoopHLSStream(HLSStream):
     metavar="STREAM_PASSWORD",
     help="The password for the stream.",
 )
+@pluginargument(
+    "afreeca-username",
+    argument_name="afreeca-username",
+    sensitive=True,
+    help=argparse.SUPPRESS,
+)
+@pluginargument(
+    "afreeca-password",
+    argument_name="afreeca-password",
+    sensitive=True,
+    help=argparse.SUPPRESS,
+)
+@pluginargument(
+    "afreeca-purge-credentials",
+    argument_name="afreeca-purge-credentials",
+    action="store_true",
+    help=argparse.SUPPRESS,
+)
+@pluginargument(
+    "afreeca-stream-password",
+    argument_name="afreeca-stream-password",
+    help=argparse.SUPPRESS,
+)
 class Soop(Plugin):
+    _OPTIONS_DEPRECATED = {
+        "afreeca-username": "username",
+        "afreeca-password": "password",
+        "afreeca-purge-credentials": "purge_credentials",
+        "afreeca-stream-password": "stream_password",
+    }
+
     _re_bno = re.compile(r"window\.nBroadNo\s*=\s*(?P<bno>\d+);")
 
     CHANNEL_API_URL = "https://live.sooplive.co.kr/afreeca/player_live_api.php"
@@ -99,6 +130,11 @@ class Soop(Plugin):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+        for opt_deprecated, opt_name in self._OPTIONS_DEPRECATED.items():
+            if (opt_value := self.options.get(opt_deprecated)) and not self.options.get(opt_name):
+                self.options.set(opt_name, opt_value)
+
         self._authed = (
             self.session.http.cookies.get("PdboxBbs")
             and self.session.http.cookies.get("PdboxSaveTicket")
```

I'll have a look at adding plugin argument aliases in the future, so plugins don't have to deal with this nonsense.

**Comment by minibox24** [comment]
image

nice

**Comment by bastimeyer** [comment]
@minibox24 Could you please quickly check and see if the authentication is working for these kinds of streams:
https://github.com/streamlink/streamlink/issues/6246#issuecomment-2414443201

If everything's working, then we can merge this.

Thanks.

**Comment by minibox24** [comment]
> @minibox24 Could you please quickly check and see if the authentication is working for these kinds of streams: [#6246 (comment)](https://github.com/streamlink/streamlink/issues/6246#issuecomment-2414443201)
> 
> If everything's working, then we can merge this.
> 
> Thanks.

I tested it, and everything works fine.

**Pull request merged**